langchain/src/schema/load.ts

```diff
@@ -0,0 +1,105 @@
+export interface SerializedFields {
+  [key: string]: unknown;
+}
+
+export interface BaseSerialized<T extends string> {
+  lc: number;
+  type: T;
+  id: string[];
+}
+
+export interface SerializedConstructor extends BaseSerialized<"constructor"> {
+  arguments: unknown[];
+  fields?: SerializedFields;
+}
+
+export interface SerializedFunction extends BaseSerialized<"function"> {
+  arguments: unknown[];
+}
+
+export interface SerializedSecret extends BaseSerialized<"secret"> {}
+
+export interface SerializedNotImplemented
+  extends BaseSerialized<"not_implemented"> {}
+
+export type Serialized =
+  | SerializedConstructor
+  | SerializedFunction
+  | SerializedSecret
+  | SerializedNotImplemented;
+
+function shallowCopy<T extends object>(obj: T): T {
+  return Array.isArray(obj) ? ([...obj] as T) : ({ ...obj } as T);
+}
+
+function replaceSecrets(
+  root: unknown[],
+  secretsMap: { [key: string]: string }
+): unknown[] {
+  const result = shallowCopy(root);
+  for (const [path, secretId] of Object.entries(secretsMap)) {
+    const [last, ...partsReverse] = path.split(".").reverse();
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let current: any = result;
+    for (const part of partsReverse.reverse()) {
+      current[part] = shallowCopy(current[part]);
+      current = current[part];
+    }
+    if (current[last] !== undefined) {
+      current[last] = {
+        lc: 1,
+        type: "secret",
+        id: [secretId],
+      };
+    }
+  }
+  return result;
+}
+
+export abstract class Serializable {
+  abstract lc_namespace: string[];
```

Comment:
No way around this eh?

Could we make it a little bit friendlier by making it recursive? e.g. `LangChainSerializationName = "structured"` and computing the full namespace by going up through the superclasses?

Does get a bit hairy if we decide to change the what a class subclasses though...

Reply:
I think generally I'd name these so that it's a bit clearer they're for serialization - a little easier for contributors to grok.

Reply:
We could make it recursive (you'll notice the lc_secrets one works like that) but the way we've organised things (eg. base classes in /schema it actually wouldn't do the right thing.